share/completions/mkdir.fish

```diff
@@ -0,0 +1,9 @@
+complete -c mkdir -s m -l mode --description 'Set file mode (as in chmod)' -x
+complete -c mkdir -s p -l parents --description 'Make parent directories as needed'
+complete -c mkdir -s v -l verbose --description 'Print a message for each created directory'
+complete -c mkdir -s Z --description 'Set SELinux security context of each created directory to the default type'
```

Comment:
>They seem to be more different than that?

Sure, but that doesn't matter here - if we have `--version`, then we have the GNU version. If we don't, then it's something else (currently it only seems to be a thing on Solaris).

>If I understand well, I should have for mkdir:

Yes, but silence `command`s stdout - `>/dev/null`.

>But, --version is only present with GNU tools, so I should also have:

Nope - the entire completion here is specific to the GNU version. The BSD version just has `-m`, `-p` and `-v` - just those short options, not the long versions.

So it's more like

```fish
if mkdir --version >/dev/null ^/dev/null
    complete -c mkdir -l version -d 'Output version'
    complete -c mkdir -s m -l mode --description 'Set file mode (as in chmod)' -x
    # [...]
else
    complete -c mkdir -s m --description 'Set file mode (as in chmod)' -x
    # [...]
end
```

In other words: Use how it reacts to "--version" as an indicator.